packages/flutter/test/material/menu_anchor_test.dart

```diff
@@ -3274,6 +3275,40 @@ void main() {
   }, skip: kIsWeb && !isCanvasKit); // https://github.com/flutter/flutter/issues/145527
 
   group('CheckboxMenuButton', () {
+    testWidgets('CheckboxMenuButton.adaptive shows the correct CheckboxMenuButton platform widget', (WidgetTester tester) async {
+      Widget buildApp(TargetPlatform platform) {
+        return MaterialApp(
+          theme: ThemeData(platform: platform),
+          home: Material(
+            child: Center(
+              child: StatefulBuilder(builder: (BuildContext context, StateSetter setState) {
+                return CheckboxMenuButton.adaptive(
+                  value: false,
+                  onChanged: (bool? newValue) {},
+                  child: const Text('adaptive checkbox'),
+                );
+              }),
+            ),
+          ),
+        );
+      }
+
+      for (final TargetPlatform platform in <TargetPlatform>[ TargetPlatform.iOS, TargetPlatform.macOS ]) {
+        await tester.pumpWidget(buildApp(platform));
+        await tester.pumpAndSettle();
+
+        expect(find.byType(CupertinoCheckbox), findsOneWidget);
```

Comment:
I would also add a check that `Checkbox` is not found. Same as in the other for loop.

Reply:
And in radio tests too.

Reply:
Adding `expect(find.byType(Checkbox), findsNothing);` within the first loop for iOS, macOS platforms fails.

I'm slightly unsure on whether this is the expected behavior, printing the widget tree shows:
```
   └ ConstrainedBox(
             BoxConstraints(0.0<=w<=18.0, 0.0<=h<=18.0),
             renderObject: RenderConstrainedBox#affd7 relayoutBoundary=up17
        )
         └ Checkbox(
             dependencies: [InheritedCupertinoTheme, _InheritedTheme, _LocalizationsScope-[GlobalKey#537ff]],
             state: _CheckboxState#3034a(tickers: tracking 4 tickers)
         )
          └ CupertinoCheckbox(
              state: _CupertinoCheckboxState#2a062(tickers: tracking 4 tickers)
          )
           └ Semantics(
...
```
If I am interpreting this correctly, it seems to suggest a parent-child relation. I'm looking through the constructor for Checkbox to understand why, but I'm not too familiar with how the widget tree is built during runtime.

Reply:
That's because `Checkbox.adaptive` is a `Checkbox` that delegates to `CupertinoCheckbox` on iOS/macOS:

https://github.com/flutter/flutter/blob/b891d2e0deee2e94a3e231193fd6c4f89fdca224/packages/flutter/lib/src/material/checkbox.dart#L487

In this case then I think it would be fine to have instead:
`expect(find.byType(Checkbox), findsOneWidget);`